### ISSUE TITLE
High CPU usage observed after upgrade to 0.6.1

### ISSUE THREAD

**Issue by namuch** [longhorn/longhorn#767]
After upgrading from 0.5.0 to 0.6.1, the instance manager pod on each node consumes much higher CPU resources.

With or without volumes mounted, the following containers are extremely busy:
-`k8s_replica-manager_instance-manager-r-xxxx...`
-`k8s_engine-manager_instance-manager-e-xxxx...`

Viewing the logs on these containers show almost no activity after initial startup and no errors.

**Comment by yasker** [comment]
Hi @namuch 

Instance managers are new components we've added to v0.6.0. Longhorn will no longer start separate pods for each engine and replica, instead instance manager will spin up processes inside it. As an extra component it will take some CPU when there is nothing running on it.

How much higher of CPU utilization you're referring to?

**Comment by namuch** [comment]
They are constantly at 4% and 6% usage when idle, determined by using docker stats.
Here is a graphing showing the usage prior to time of upgrade until current time:

![image](https://user-images.githubusercontent.com/18311962/65619130-73973300-df74-11e9-8110-e050761158c7.png)

**Comment by yasker** [comment]
@namuch Thanks for reporting. I think there is room for improvement. We will see what can we do about it for the v0.7.0 release.

**Comment by yasker** [comment]
Before the fix, on a three-node cluster, each with 2 VCPUs, Longhorn just installed:
```
$ kl top pod
NAME                                        CPU(cores)   MEMORY(bytes)
csi-attacher-799cdf866b-jpjhn               2m           11Mi
csi-attacher-799cdf866b-z7wn4               1m           4Mi
csi-attacher-799cdf866b-z8mrc               1m           4Mi
csi-provisioner-7f4bccb59-4hgrj             2m           7Mi
csi-provisioner-7f4bccb59-dfd6k             1m           4Mi
csi-provisioner-7f4bccb59-mpdlh             1m           4Mi
engine-image-ei-7c102650-99mzx              4m           11Mi
engine-image-ei-7c102650-p6pjr              4m           1Mi
engine-image-ei-7c102650-v4zdm              4m           1Mi
instance-manager-e-156b9d2c                 46m          4Mi
instance-manager-e-65b697d2                 53m          5Mi
instance-manager-e-dcf4ba5a                 43m          3Mi
instance-manager-r-1b60884d                 49m          3Mi
instance-manager-r-b88c3039                 48m          3Mi
instance-manager-r-c7e617c5                 50m          4Mi
longhorn-csi-plugin-gtkch                   0m           9Mi
longhorn-csi-plugin-ns25b                   1m           9Mi
longhorn-csi-plugin-zxx5k                   1m           9Mi
longhorn-driver-deployer-7b6b5f564c-trzrr   1m           8Mi
longhorn-manager-fj5cp                      7m           20Mi
longhorn-manager-mp557                      17m          21Mi
longhorn-manager-rj747                      11m          22Mi
longhorn-ui-b5f866f98-7jz95                 1m           6Mi

$ kl top node
NAME                    CPU(cores)   CPU%   MEMORY(bytes)   MEMORY%
yasker-longhorn-dev-1   1479m        73%    1906Mi          49%
yasker-longhorn-dev-2   1526m        76%    2224Mi          57%
yasker-longhorn-dev-3   1423m        71%    2151Mi          55%
```
After the fix:
```
$ kl top pod
NAME                                        CPU(cores)   MEMORY(bytes)
csi-attacher-799cdf866b-jpjhn               2m           10Mi
csi-attacher-799cdf866b-z7wn4               1m           4Mi
csi-attacher-799cdf866b-z8mrc               1m           4Mi
csi-provisioner-7f4bccb59-4hgrj             2m           7Mi
csi-provisioner-7f4bccb59-dfd6k             1m           4Mi
csi-provisioner-7f4bccb59-mpdlh             1m           4Mi
engine-image-ei-5d7540ae-44h2h              4m           15Mi
engine-image-ei-5d7540ae-8xhm9              4m           14Mi
engine-image-ei-5d7540ae-pgkcp              4m           13Mi
instance-manager-e-ae5a3054                 7m           6Mi
instance-manager-e-bb13bf55                 4m           6Mi
instance-manager-e-c20c1d59                 6m           6Mi
instance-manager-r-324f154a                 6m           6Mi
instance-manager-r-a0ffc9d8                 6m           6Mi
instance-manager-r-e20e17ed                 6m           6Mi
longhorn-csi-plugin-gtkch                   0m           9Mi
longhorn-csi-plugin-ns25b                   1m           9Mi
longhorn-csi-plugin-zxx5k                   1m           9Mi
longhorn-driver-deployer-7b6b5f564c-trzrr   1m           8Mi
longhorn-manager-4pnjq                      8m           25Mi
longhorn-manager-b2fbx                      5m           23Mi
longhorn-manager-bkch6                      11m          26Mi
longhorn-ui-b5f866f98-7jz95                 1m           6Mi

$ kl top node
NAME                    CPU(cores)   CPU%   MEMORY(bytes)   MEMORY%
yasker-longhorn-dev-1   329m         16%    1790Mi          46%
yasker-longhorn-dev-2   367m         18%    2197Mi          57%
yasker-longhorn-dev-3   365m         18%    2144Mi          55%
```

Noted we need to wait for a few minutes for `kubectl top` to update the data.

**Comment by sybnex** [comment]
So when can we expect the new release with this fix? The kubernetes cluster is really unusable at the moment.

It would be great to have this as soon as possible.

**Comment by yasker** [comment]
@sybnex We're planning to releasing the fix with v0.6.2 by the end of next week. There are a few other bugs we've found and need to address before the release.

**Comment by meldafrawi** [comment]
Validation: **PASSED**
